### ISSUE TITLE
New rule: avoid_similar_names

### ISSUE THREAD

**Issue by danylo-safonov-solid** [solid-software/solid_lints#95]
e.g. if names differ by one character, especially if the characters are similar (e.g. digit 1 and letter l)

BAD:
```dart
bool isEqual(SomeClass someClass1, SomeClass someClass2) ...
```

BETTER:
```dart
bool isEqual(SomeClass someClassA, SomeClass someClassB) ...
```

GOOD:
```dart
bool isEqual(SomeClass that, SomeClass other) ...
```

GOOD:
```dart
bool isEqual(SomeClass first, SomeClass second) ...
```

**Comment by solid-illiaaihistov** [comment]
Implemented, closing the task.
